internal/service/lightsail/database.go

```diff
@@ -0,0 +1,462 @@
+package lightsail
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lightsail"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+)
+
+func ResourceDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: ResourceDatabaseCreate,
+		Read:   ResourceDatabaseRead,
+		Update: ResourceDatabaseUpdate,
+		Delete: ResourceDatabaseDelete,
+		Importer: &schema.ResourceImporter{
+			State: ResourceDatabaseImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"master_database_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"master_password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"master_username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"blueprint_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"bundle_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			// Optional attributes
+			"preferred_backup_window": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"preferred_maintenance_window": {
```

Comment:
Please add input validation to all schema attributes. This is to provide `terraform plan` validation to proactively alert the user to known invalid input.

Reply:
I added Input validation on all the fields that made sense to have validation. I also added acceptance testing to ensure the validations are working as expected.